File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """
            2 ptr

            1   2   3   4
            |           |
        """

        n = len(nums)
        left, right = 0,n-1

        while left < right:
            s = nums[left] + nums[right]

            if s == target:
                return [left+1, right+1]
            
            elif s > target:
                right -= 1
            
            else:
                left += 1
        
        return []

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        /*
            3   4   5   6   1   2
                    |

        */

        int n = nums.size();
        int left=0, right=n-1;

        while(left < right){
            int mid = left + (right-left)/2;
            if(nums[right] >= nums[mid]){
                if(nums[right] >= nums[left]){
                    right--;
                }
                else{
                    left++;
                }
            }
            else{
                if(nums[mid] >= nums[left]){
                    left++;
                }
                else{
                    right--;
                }
            }
        }
        return nums[left];
    }
};
